src/runtime/pythonengine.cs

```diff
@@ -538,6 +546,56 @@ public static GILState GIL()
 
             return new GILState();
         }
+        
+        private static Dictionary<string, PyScope> NamedScopes = new Dictionary<string, PyScope>();
+
+        public static PyScope CreateScope()
+        {
+            return CreateScope(null);
+        }
+
+        public static PyScope CreateScope(string name)
+        {
+            if (name != null && NamedScopes.ContainsKey(name))
+            {
+                throw new PyScopeException(String.Format("ScopeStorage '{0}' has existed", name));
+            }
+            var scope = new PyScope(name);
+            if (name != null)
```

Comment:
Empty string means the scope will not be registered. It should be used and discarded immediately, or it was save in a class field.

Named scope can be shared just like a python module.
Now we have two methods for scope sharing, ImportScope and ImportAllFromScope.